controller/controller.go

```diff
@@ -684,7 +684,7 @@ func (ctrl *ProvisionController) processNextClaimWorkItem() bool {
 		}
 
 		if err := ctrl.syncClaimHandler(key); err != nil {
-			if ctrl.claimQueue.NumRequeues(obj) < ctrl.failedProvisionThreshold {
+			if ctrl.claimQueue.NumRequeues(obj) < ctrl.failedProvisionThreshold || ctrl.failedProvisionThreshold == 0 {
 				glog.Warningf("Retrying syncing claim %q because failures %v < threshold %v", key, ctrl.claimQueue.NumRequeues(obj), ctrl.failedProvisionThreshold)
```

Comment:
this message will be very confusing with `failedProvisionThreshold==0`.

Reply:
new patch, PTAL